crates/isograph_config/src/compilation_options.rs

```diff
@@ -98,6 +99,20 @@ pub enum JavascriptModule {
     EsModule,
 }
 
+#[derive(Debug, Clone)]
+pub struct PersistedDocumentsOptions {
+    pub directory: Option<AbsolutePathAndRelativePath>,
+    pub filename: ArtifactFileName,
```

Comment:
`artifact_directory` uses a pathbuf, e.g. `pub artifact_directory: Option<PathBuf>,`, can we do the same here?

Reply:
I don't think any of these need to be required, e.g. perhaps it should default to `__isograph/persistedDocuments.json` if `filename` is missing?

Reply:
There is another struct ConfigFilePersistedDocumentsOptions where directory is an Option and filename is initialized with default value.

Reply:
Sorry, to clarify, I'm suggesting we make it a single `Option<PathBuf>` in both `PersistedDocumentsOptions` and `ConfigFilePersistedDocumentsOptions`, and that pathbuf is relative to the `__isograph` directory

Reply:
Do you mean we should have a single option instead of both directory and filename? I thought we might need one day a different serialization strategy, e.g. hot chocolate checks the specified folder for files in the following format: {Hash}.graphql.

Reply:
That's a good idea to support that eventually. Once we do that, it would probably be best to add a discriminator, i.e. have the `persisted_operations` object have the following type (in JS)
```
{ kind: 'OneJson', file, algorith } | { kind: 'FolderContainingHashedFiles', folder, algorith }
```

I'm okay with backwards-incompatible changes here. (And really, in general. I don't think users can expect stability until at least we have all the primitives figured out.)

Reply:
yeah